From ec084ad18bd422d114f6cbfe4e7d9d951a658188 Mon Sep 17 00:00:00 2001
From: Yadong Qi <yadong.qi@intel.com>
Date: Wed, 7 Jul 2021 16:38:03 +0800
Subject: [PATCH] prebuilt: Allow adbd to use vsock_socket

Align policy in prebuilt/adbd.te to pass
the sepolicy_freeze_test.

Signed-off-by: Yadong Qi <yadong.qi@intel.com>
---
 prebuilts/api/30.0/private/adbd.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/prebuilts/api/30.0/private/adbd.te b/prebuilts/api/30.0/private/adbd.te
index e4498b05b..5d402e81c 100644
--- a/prebuilts/api/30.0/private/adbd.te
+++ b/prebuilts/api/30.0/private/adbd.te
@@ -44,6 +44,9 @@ dontaudit adbd self:global_capability_class_set sys_resource;
 # this occurs. (b/123569840)
 dontaudit adbd self:{ socket vsock_socket } create;
 
+# Allow adbd inside vm to forward vm's vsock.
+allow adbd self:vsock_socket { create_socket_perms_no_ioctl listen accept };
+
 # Create and use network sockets.
 net_domain(adbd)
 
-- 
2.25.1

